### PR TITLE
Add `TSDataset.create_from_misaligned` constructor 

### LINKED ISSUE #261: Add a new constructor into `TSDataset` to work with unaligned data
### 🚀 Feature Request

We could add a new constructor for handling unaligned data.

### Proposal

Add classmethod `TSDataset.create_aligned_ts` which should handle alignment and creation:
- Parameters
  - `df: pd.DataFrame`: df in a long format
  - `freq: Optional[str]`: freq in initial `df`
  - `df_exog: pd.DataFrame`: `df_exog` in a long format
  - `known_future` as in `__init__`
  - `future_steps: int`: number of steps to extend timestamp into the future in `original_timestamp_name`
  - `original_timestamp_name: str = "external_timestamp"`
- Algorithm
  - Use `etna.datasets.utils.infer_alignment` to infer alignment in `df`
  - Align `df` using `etna.datasets.utils.apply_alignment` and remove existing timestamp
  - Align `df_exog` using `etna.datasets.utils.apply_alignment` and remove existing timestamp
  - Create df with timestamps using `etna.datasets.utils.make_timestamp_df`
  - Concatenate `df_exog` and `timestamp_df`
  - Create `TSDataset` using `__init__`
- We doesn't accept `hierarchical_structure` as it doesn't make much sense for realigned data
- Fail if there is name collision with `original_timestamp_name` in `df_exog`

### Test cases

- Add tests on `TSDataset.create_aligned_ts`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by d-a-bunin**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look at #261.

## Closing issues
Closes #261.

### CODE REVIEW

File: tests/test_datasets/test_dataset.py [hunk 1]
```diff
@@ -1684,3 +1687,193 @@ def test_check_timestamp_type_warning():
...
+
+
+"""TODO: что хотим проверить
+- ts.freq is None
+- ts.known_future содержит original_timestamp_name
+- ts.df_exog содержит original_timestamp_name
+- ts.df_exog получается путем join-а исходного df_exog и результата make_timestamp_df
+- ts.raw_df получается путем alignment-а
+- ts.df_exog -- коллизия имени с original_timestamp_name
+"""
```
Comment (ostreech1997) [inline tests/test_datasets/test_dataset.py:1699]: We should delete this section.

File: tests/test_datasets/test_dataset.py [hunk 2]
```diff
@@ -1684,3 +1687,193 @@ def test_check_timestamp_type_warning():
...
+    df = request.getfixturevalue(df_name)
+    ts = TSDataset.create_from_misaligned(
+        df=df, df_exog=None, freq=freq, original_timestamp_name=original_timestamp_name, future_steps=future_steps
+    )
+
+    alignment = infer_alignment(df)
+    expected_raw_df = TSDataset.to_dataset(apply_alignment(df=df, alignment=alignment))
+    pd.testing.assert_frame_equal(ts.raw_df, expected_raw_df)
+
+    timestamp_df = make_timestamp_df_from_alignment(
```
Comment (ostreech1997) [inline tests/test_datasets/test_dataset.py:1724]: Why do we check it in this test?

File: tests/test_datasets/test_dataset.py [hunk 3]
```diff
@@ -1684,3 +1687,193 @@ def test_check_timestamp_type_warning():
...
+        ("df_misaligned_datetime", "df_exog_aligned_datetime", "D", ["exog_1"], "external_timestamp", 1),
+        ("df_misaligned_datetime", "df_exog_misaligned_datetime", "D", ["exog_1"], "external_timestamp", 1),
+        ("df_misaligned_int", "df_exog_aligned_int", None, ["exog_1"], "external_timestamp", 1),
+        ("df_misaligned_int", "df_exog_misaligned_int", None, ["exog_1"], "external_timestamp", 1),
+        ("df_misaligned_datetime", "df_exog_misaligned_datetime", "D", ["exog_1"], "new_timestamp", 1),
+        ("df_misaligned_datetime", "df_exog_misaligned_datetime", "D", ["exog_1"], "external_timestamp", 3),
+        ("df_misaligned_datetime", "df_exog_misaligned_datetime", "D", ["exog_1"], "external_timestamp", 100),
+    ],
+)
+def test_create_from_misaligned_with_exog(
```
Comment (ostreech1997) [inline tests/test_datasets/test_dataset.py:1754]: Is it fine, that we check `df_exog` and `raw_df` in the same test?

Comment (d-a-bunin) [reply tests/test_datasets/test_dataset.py:1754]: We check all parameters in the same test. I doesn't see any benefit in separating it right now.

Comment (ostreech1997) [reply tests/test_datasets/test_dataset.py:1744]: You're right, if we split it into two separate tests, then we get duplication of code.

**Review by ostreech1997 — APPROVED** [review]

**Pull request merged**